### ISSUE TITLE
end to end tests for multicluster rollouts

### ISSUE THREAD

**Issue by droot** [GoogleContainerTools/kpt#3776]
This is an umbrella issue for tracking end to end testing of rollouts scenario. To begin with, following should be supported:

1. Cluster add/delete
2. Different rollout strategy
3. Package (and variants) rollout/rollback (1 package to multiple cluster, variant per cluster)
4. Scale testing

We would like to support following environments:

1. Using `kind` clusters that one can test for dev purpose
2. `GKEFleet` based clusters that we can test out periodically.

**Comment by droot** [comment]
This PR https://github.com/GoogleContainerTools/kpt/pull/3879 adds following tests:

1. Package is deployed to specified one cluster among the two target clusters
2. Package is deployed to two clusters through progressive strategy and it also covers the manual advancement scenario.

The linked PR also adds support for bringing target environment and I think with very little changes dynamic add/deletion of target clusters can also be added.

**Comment by droot** [comment]
We should create a separate issue to track the remaining end to end tests once the linked PR is merged.

**Comment by droot** [comment]
Revisited the current end to end tests, and I think we have sufficient coverage that we wanted to tackle in MVP, so closing this.
